packages/core/src/sync-historical/service.bench.ts

```diff
@@ -0,0 +1,53 @@
+import { setupAnvil, setupContext, setupSyncStore } from "@/_test/setup.js";
+import { simulate } from "@/_test/simulate.js";
+import { publicClient } from "@/_test/utils.js";
+import { type TestContext, bench } from "vitest";
+import { HistoricalSyncService } from "./service.js";
+
+let context = {} as TestContext;
+const setup = async () => {
+  context = {} as TestContext;
+  setupContext(context);
+  await setupAnvil(context);
+
+  for (let i = 0; i < 100; i++)
+    await simulate({
+      erc20Address: context.erc20.address,
+      factoryAddress: context.factory.address,
+    });
+};
+
+const getBlockNumbers = () =>
+  publicClient.getBlockNumber().then((b) => ({
+    latestBlockNumber: Number(b) + 5,
+    finalizedBlockNumber: Number(b),
+  }));
+
+bench(
+  "Historical sync benchmark",
+  async () => {
+    const teardownSync = await setupSyncStore(context);
```

Comment:
So, you have to do it this way because `vitest bench` doesn't allow `beforeEach` ?

Reply:
yes